language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -49,6 +51,7 @@ data Name a = Name { nameAttribute :: a
 newtype TyName a = TyName { unTyName :: Name a }
     deriving (Show, Generic, Lift)
     deriving newtype (Eq, Ord, Functor, NFData)
+instance Wrapped (TyName a)
```

Comment:
I'm not a fan of this change. Now you need to provide an instance for each newtype forever, that's just boilerplate. And you need to enable either `GADTs` or `TypeFamilies` for this, boilerplate again. And type signatures did not become shorter:

```
-    :: (Newtype new, HasUnique (Unwrap new) unique', Coercible unique' unique)
+    :: (Wrapped new, HasUnique (Unwrapped new) unique', Coercible unique' unique)
```

This type class is only needed for wrapping/unwrapping non-newtypes (I suppose in can handle a non-existential GADT with a single constructor). Do we really need this generality at the cost of writing more boilerplate?

Reply:
I don't really think we're writing much boilerplate. We're going to use this maybe two or three times. The main thing I like about this is just that we're no longer using our own code, which just makes me more confident that it's right.

Reply:
I don't think safe coercions can be wrong at all. Well, I do not hate this change, I just like the previous version more, but if you prefer to use the `lens`' one, that makes sense too.

Reply:
We're still using the coercions! The only thing that's not there is the clever thing where you work out that something is a newtype without having to say anything.

Reply:
> We're still using the coercions!

Sure, I responded to:

> makes me more confident that it's right.